datalab/datalab_session/data_operations/data_operation.py

```diff
@@ -50,6 +52,29 @@ def wizard_description():
             for the frontend to create custom input widgets for it in a wizard
         """
 
+    def _validate_inputs(self, input_key='input_files', minimum_inputs=1):
+        """ Validate the inputs for the operation.
+            This method should raise an exception if the inputs are not valid.
+            It should return a list of input files or data that will be used in the operate() method.
+        """
+        input_list = self.input_data.get(input_key, [])
+        if not input_list or len(input_list) < minimum_inputs:
+            raise ClientAlertException(f'Operation {self.name()} requires at least {minimum_inputs} input file(s).')
+        
+        return input_list
+    
+    def _process_inputs(self, submitter, input_list, input_processing_progress=0.1):
```

Comment:
We should get rid of this I think. It doesn't really do much, and it actually breaks the per-image loading for subtraction and normalization operations that don't need to load all the data into memory at once. I think we are better off creating the InputDataHandlers where we want them within each operation, since that step explicitly uses a bunch of memory and potentially downloads the file as well.